### PR TITLE
ci: Allow build action to be triggered remotely

### PR DESCRIPTION
**Pull request by DaniBodor**
This change allows the build action to be triggered by an action in a remote repo. 
This will be used by [eitprocessing](https://github.com/EIT-ALIVE/eitprocessing) as of [PR 235](https://github.com/EIT-ALIVE/eitprocessing/pull/235), and these PRs should be reviewed together.

EDIT: @wbaccinelli , i had tagged you for review, but have since realized that it wasnt working as intended. I will fix it first and then request a new review when it's working.
--> THIS IS NOW DONE!

Some additional changes were made to the actions. All commits are atomic and can be reviewed independently.
Main additional changes:
- build actions explicitly use os and python versions from matrix
- lint action was moved to separate file (it now has very little in common with main build action and otherwise i would need to specify extra conditions to avoid it being triggered remotely)
- a dedicated action that can be triggered remotely was also added. This can be useful for trying out things in future updates. Not sure it should be kept or deleted.

### CODE REVIEW

File: .github/workflows/remote_build.yml
Comment (wbaccinelli) [inline .github/workflows/remote_build.yml:1]: It's nice to have a debug workflow ready to be used in case of need

**Review by wbaccinelli — APPROVED** [review]
Good job! Comments are minor.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -41,28 +41,31 @@ jobs:
         run: >
           docker run
           --rm
-          -v ${{ github.workspace }}:/ci
+          -v $GITHUB_WORKSPACE:/ci
```
[the remark this answers is not in the document]
Comment (DaniBodor) [reply .github/workflows/build.yml:53]: I think nothing (it came up when I was checking things with chatgpt), but found this slightly more readable. Wouldn't have made a separate PR for it, but while I was at it...

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Python info
+        shell: bash -e {0}
+        run: |
+          which python3
+          python3 --version
```
[the remark this answers is not in the document]
Comment (DaniBodor) [reply .github/workflows/lint.yml:42]: no... it was already in there. I guess for troubleshooting reasons.

**Pull request merged**